cogs5e/sheets/dicecloud.py

```diff
@@ -129,6 +133,27 @@ def get_stats(self) -> BaseStats:
         self.stats = stats
         return stats
 
+    def get_coinpurse(self):
+        """ Due to the format of the coin handling in DiceCloud, this is going to be slightly messy."""
+        if self.character_data is None: raise Exception('You must call get_character() first.')
+        if self.coinpurse:
+                return self.coinpurse
+        character = self.character_data
+
+        coins = {
+            "pp": 0,
+            "gp": 0,
+            "ep": 0,
+            "sp": 0,
+            "cp": 0
+        }
+
+        for i in self.character_data.get('items'):
+            if re.search(r"^((plat|gold|electrum|silver|copper)( piece)?|(pp|gp|ep|sp|cp))", i['name'], re.IGNORECASE):
```

Comment:
```suggestion
            if re.fullmatch(r"(platinum|gold|electrum|silver|copper)( pieces?)?|(pp|gp|ep|sp|cp)", i['name'], re.IGNORECASE):
```

Reply:
Due to the fact that you can create your own, and the wording on dice cloud, I opted to use plat and piece over these.

If someone created a platinum pieces, I can see them setting it as plat.

Also, the item by default is called gold pieces.  However, in the item add screen it asks for name + plural name, so I wanted to be able to pickup singular name.


@zhudotexe

Reply:
We can use `r"(plat(inum)?|gold|electrum|silver|copper)( pieces?)?|(pp|gp|ep|sp|cp)"` to match both `plat` and `platinum`. The `s?` handles 0 or 1 `s`, so this will handle both the singular and plural case.